system-x/services/tekton/src/main/java/software/tnb/tekton/service/Tekton.java

```diff
@@ -0,0 +1,120 @@
+import software.tnb.tekton.validation.TektonValidation;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.auto.service.AutoService;
+
+import java.io.IOException;
+import java.util.function.Predicate;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.tekton.client.TektonClient;
+
+@AutoService(Tekton.class)
+public class Tekton extends Service<NoAccount, TektonClient, TektonValidation> implements OpenshiftDeployable, WithOperatorHub {
+    private static final Logger LOG = LoggerFactory.getLogger(Tekton.class);
+
+    public Tekton() {
+
+    }
+
+    protected TektonClient client() {
+        if (OpenshiftClient.get().isAdaptable(TektonClient.class)) {
+            return OpenshiftClient.get().adapt(TektonClient.class);
+        }
+        throw new IllegalArgumentException("Unable to adapt OpenshiftClient to TektonClient");
+    }
+
+    @Override
+    public TektonValidation validation() {
+        if (validation == null) {
+            validation = new TektonValidation(client());
+        }
+        return validation;
+    }
+
+    public void create() {
+        LOG.debug("Creating Tekton operator");
+        createSubscription();
+    }
+
+    public void undeploy() {
+        LOG.debug("Undeploy Tekton operator");
+        CustomResourceDefinitionContext crd = new CustomResourceDefinitionContext.Builder()
+            .withName("tektonconfigs.operator.tekton.dev")
+            .withGroup("operator.tekton.dev")
+            .withVersion("v1alpha1")
+            .withPlural("tektonconfigs")
+            .withScope("Cluster")
+            .build();
+
+        try {
+            OpenshiftClient.get().customResource(crd).delete("openshift-pipelines", "config", true);
+            WaitUtils.waitFor(() ->
+                    OpenshiftClient.get().namespaces().withName("openshift-pipelines").get() == null,
+                "Waiting until the openshift-pipelines namespace is removed");
+            deleteSubscription(() -> OpenshiftClient.get().getLabeledPods("name", "openshift-pipelines-operator")
+                .stream().noneMatch(p -> p.getMetadata().getName().contains("openshift-pipelines-operator")));
+        } catch (IOException e) {
+            LOG.error("Error on Tekton deletetion", e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void openResources() {
+        // no-op
+    }
+
+    @Override
+    public void closeResources() {
+        // no-op
+    }
+
+    @Override
+    public boolean isReady() {
+        if (OpenshiftClient.get().namespaces().withName("openshift-pipelines").isReady()) {
+            return OpenshiftClient.get().pods().inNamespace("openshift-pipelines").list().getItems().size() == 12
+                && OpenshiftClient.get().apps().deployments().list().getItems().stream().allMatch(p -> p.getStatus().getReadyReplicas() == 1);
```

Comment:
should this check the deployments in tnb namespace or in `openshift-pipelines` namespace? as currently it checks it in tnb namespace